packages/keyring/src/pair/index.ts

```diff
@@ -173,9 +173,8 @@ export function createPair ({ toSS58, type }: Setup, { publicKey, secretKey }: P
       );
     },
     toJson: (passphrase?: string): KeyringPair$Json => {
-      // NOTE: in the case of Ethereum and Ecdsa, we encode the ful publicKey - to allow extraction. This will NOT match
-      // e.g. an Ethereum address which is non-reversible, so not used here)
-      const address = ['ecdsa', 'ethereum'].includes(type)
+      // For ecdsa and ethereum, address and publicKey are not the same (and not bijective). We want the publicKey here
+      const address = ['ethereum'].includes(type)
```

Comment:
We have dropped one. (It has the same issue, hence the original check and the check in master)

Reply:
fixed